CMakeLists.txt

```diff
@@ -399,74 +399,154 @@ install (TARGETS pwsafe RUNTIME DESTINATION "bin")
-        IMMEDIATE @ONLY)
+  configure_file(
+      "${CMAKE_CURRENT_SOURCE_DIR}/install/cmake_uninstall.cmake.in"
+      "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
+      IMMEDIATE @ONLY)
 
-    add_custom_target(uninstall
-        COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
+  add_custom_target(uninstall
+      COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake)
 endif()
 
 
-# Packaging stuff:
-## Common:
+###################################################################################################
+#
+# P A C K A G I N G
+#
+###################################################################################################
+
+#
+# Major version number of software version (e.g. 1.08-BETA).
+#                                                ^
 set (CPACK_PACKAGE_VERSION_MAJOR "${pwsafe_VERSION_MAJOR}")
-set (CPACK_PACKAGE_VERSION_MAJOR "${pwsafe_VERSION_MINOR}")
+
+#
+# Minor version number of software version (e.g. 1.08-BETA), which should consist of two digits.
+#                                                   ^
+string(LENGTH ${pwsafe_VERSION_MINOR} VERSION_MINOR_LENGTH)
+if (VERSION_MINOR_LENGTH GREATER 1)
+  set (CPACK_PACKAGE_VERSION_MINOR "${pwsafe_VERSION_MINOR}")
+else ()
+  set (CPACK_PACKAGE_VERSION_MINOR "0${pwsafe_VERSION_MINOR}")
+endif ()
+
+#
+# Mandatory part is complete. What follows is optional.
+#
+set (CPACK_PACKAGE_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}")
+
+#
+# Revision number of software version (e.g. 1.08.1-BETA).
+#                                                ^
 if (pwsafe_REVISION)
-   set (CPACK_PACKAGE_VERSION_PATCH "${pwsafe_REVISION}")
+  set (CPACK_PACKAGE_VERSION_PATCH "${pwsafe_REVISION}")
+  set (CPACK_PACKAGE_VERSION "${CPACK_PACKAGE_VERSION}.${CPACK_PACKAGE_VERSION_PATCH}")
 endif (pwsafe_REVISION)
+
+#
+# Special build tag of software version (e.g. 1.08.1-BETA).
+#                                                    ^
 if (pwsafe_SPECIALBUILD)
-   set (CPACK_PACKAGE_VERSION "${pwsafe_VERSION}-${pwsafe_SPECIALBUILD}")
-else ()
-   set (CPACK_PACKAGE_VERSION "${pwsafe_VERSION}")
+  set (CPACK_PACKAGE_VERSION "${CPACK_PACKAGE_VERSION}-${pwsafe_SPECIALBUILD}")
 endif (pwsafe_SPECIALBUILD)
+
+#
+# Miscellaneous package information.
+#
 set (CPACK_PACKAGE_CONTACT "Rony Shapiro <ronys@pwsafe.org>")
 set (CPACK_PACKAGE_VENDOR "Rony Shapiro")
 set (CPACK_PACKAGE_DESCRIPTION_SUMMARY "Password Safe is a password database utility.")
 set (CPACK_RESOURCE_FILE_LICENSE "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE")
 set (CPACK_PACKAGE_DESCRIPTION_FILE "${CMAKE_CURRENT_SOURCE_DIR}/install/description.txt")
 set (CPACK_STRIP_FILES "TRUE")
 
+#
+# Package file name is project name.
+#
+if (NOT CPACK_PACKAGE_NAME)
+  string(TOLOWER "${PROJECT_NAME}" CPACK_PACKAGE_NAME)
+endif ()
+
+#
+# Determine the architecture (amd64, i386) with command 'dpkg --print-architecture'
```

Comment:
'dpkg' isn't part of the RPM-based distros. A more portable way to do this is via the 'arch' command or 'uname -m'.

Do you have the ability to test this pull request on Fedora or CentOS?

Reply:
My goal was to get 'amd64' as in the case of the already published packages. This was only provided by 'dpkg'. 'arch' and 'uname -m' provide 'x86_64' instead of 'amd64'. However, 'x86_64' could be easily replaced by 'amd64'. Please let me know which identifier is desired.
I would very appreciate if you could do the testing on Fedora or CentOS.

Reply:
I'd like to map x86_64 as returned by 'uname -m' (more portable that arch) to arm64.

I'll do the work on RPM distros once you're done with this part.